packages/2d/src/components/Polygon.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  // PossibleSpacing,
+  BBox,
+  // SpacingSignal,
+} from '@motion-canvas/core/lib/types';
+import {Shape, ShapeProps} from './Shape';
+import {initial, signal} from '../decorators';
+import {SignalValue, SimpleSignal} from '@motion-canvas/core/lib/signals';
+
+function drawPolygon(
+  path: CanvasRenderingContext2D | Path2D,
+  rect: BBox,
+  // width: number,
+  // height: number,
+  sides: number,
+) {
+  const width = rect.width / 2;
+  const height = rect.height / 2;
+  for (let i = 0; i <= sides; i += 1) {
+    const theta = (i * 2 * Math.PI) / sides;
+    const x = width * Math.sin(theta);
+    const y = -height * Math.cos(theta);
+    if (i == 0) {
+      path.moveTo(x, y);
+    } else {
+      path.lineTo(x, y);
+    }
+  }
+  path.closePath();
```

Comment:
This is an optional suggestion, but I think the code would be a bit more readable when using vectors:
```suggestion
  const size = rect.size.scale(0.5);
  for (let i = 0; i <= sides; i++) {
    const theta = (i * 2 * Math.PI) / sides;
    const direction = Vector2.fromRadians(theta).perpendicular;
    const vertex = direction.mul(size);
    if (i == 0) {
      moveTo(path, vertex);
    } else {
      lineTo(path, vertex);
    }
  }
  path.closePath();
  ```
  since they better illustrate the math that's happening

Reply:
Thank you.  This is helpful.

I had to make a small modification to get this to run:

``` js
    if (i == 0) {
      path.moveTo(vertex.x, vertex.y);
    } else {
      path.lineTo(vertex.x, vertex.y);
    }
```

Reply:
The `moveTo` and `lineTo` functions are helper functions that can be imported from `@motion-canvas/2d/lib/utils`. These functions actually accept vectors as inputs instead of expecting each point individually.

Reply:
I tried that and couldn't get it to work.  I'll try again.

Reply:
Can you provide more details on what exactly wasn't working?

Reply:
OK, I figured out why.  I haven't pulled the latest changes from `main`, and now I see there are changes in v3.1 that allows this new syntax.
Thank you.

Reply:
Hard to say now.  Probably user error.

(I thought something changed in v3.1, so I did a `git pull` and tried again. This time it worked. So if there were no changes in `utils` in v3.1, then this was certainly user error.)